src/Core/View.js

```diff
@@ -217,11 +234,14 @@ function _preprocessLayer(view, layer, provider) {
  * });
  *
  * // Add Elevation Layer
- * view.addLayer({
+ * var layer = view.addLayer({
  *      type: 'elevation',
  *      id: 'iElevation',
  * });
  *
+ * // React to preprocessing
+ * layer.whenReady.then(() => { ... });
```

Comment:
An alternative would be to resolve whenReady with the layer. 

It does not solve the use case of needing the layer object before it is ready though. Querying `view.getLayers` just after `view.addLayer` seems cumbersome to me. Do you think this use case will be often encountered? If not, I guess we can indeed return the promise.